inference-engine/thirdparty/movidius/XLink/shared/src/XLinkSemaphore.c

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "XLinkSemaphore.h"
+#include "XLinkErrorUtils.h"
+#include "XLinkLog.h"
+
+static pthread_mutex_t ref_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t ref_cond = PTHREAD_COND_INITIALIZER;
+
+int XLink_sem_inc(XLink_sem_t* sem)
+{
+    XLINK_RET_IF_FAIL(pthread_mutex_lock(&ref_mutex));
+    int ret = 0;
+    if (sem->refs < 0) {
```

Comment:
I think we need to check that almost in all functions. No?
Why no `XLINK_RET_IF_FAIL`?

Reply:
> I think we need to check that almost in all functions. No?

Added in the last commit

> Why no XLINK_RET_IF_FAIL?

We still need to unlock mutex